src_kt/javatests/arcs/storage/StoreTest.kt

```diff
@@ -0,0 +1,354 @@
+    val (driver, _) = setupMocks()
+    val receiverCaptor = argumentCaptor<suspend (CrdtCount.Data, Int) -> Unit>()
+    val firstCallComplete = CompletableDeferred<Unit>(coroutineContext[Job.Key])
+    val secondCallComplete = CompletableDeferred<Unit>(coroutineContext[Job.Key])
+
+    whenever(driver.registerReceiver(anyOrNull(), receiverCaptor.capture())).thenReturn(Unit)
+    whenever(driver.send(any(), any())).then {
+      firstCallComplete.complete(Unit)
+      return@then false
+    }
+
+    val activeStore = createStore().activate()
+
+    // local count from proxy
+    val count = CrdtCount()
+    count.applyOperation(Increment("me", 0 to 1))
+
+    // conflicting remote count from store
+    val remoteCount = CrdtCount()
+    remoteCount.applyOperation(Increment("them", 0 to 1))
+
+    val result = activeStore.onProxyMessage(ProxyMessage.ModelUpdate(count.data, 1))
+    println("Received result.")
+
+    firstCallComplete.await()
+    assertThat(result).isTrue()
+
+    println("Setting up for round two")
+    // Reset, this time we'll capture the model it receives.
+    val modelCaptor = argumentCaptor<CrdtCount.Data>()
+    whenever(driver.send(modelCaptor.capture(), any())).then {
+      secondCallComplete.complete(Unit)
+      return@then true
+    }
+
+    println("Calling receiver: ${receiverCaptor.lastValue}")
+    receiverCaptor.lastValue(remoteCount.data, 1)
+    println("Called captor")
+
+    secondCallComplete.await()
+
+    count.merge(remoteCount.data)
+    assertThat(modelCaptor.lastValue).isEqualTo(count.data)
+  }
+
+  @Test
+  fun resolves_combinationOfMessages_fromProxyAndDriver() = runBlockingTest {
+    val (driver, _) = setupMocks()
+    val receiverCaptor = argumentCaptor<suspend (CrdtCount.Data, Int) -> Unit>()
+    val driverModelCaptor = argumentCaptor<CrdtCount.Data>()
+    whenever(driver.registerReceiver(anyOrNull(), receiverCaptor.capture())).thenReturn(Unit)
+    whenever(driver.send(driverModelCaptor.capture(), any())).thenReturn(true)
+
+    val activeStore = createStore().activate() as DirectStore
+
+    activeStore.onProxyMessage(ProxyMessage.Operations(listOf(Increment("me", 0 to 1)), id = 1))
+    activeStore.onProxyMessage(ProxyMessage.Operations(listOf(Increment("me", 1 to 2)), id = 1))
+    activeStore.onProxyMessage(ProxyMessage.Operations(listOf(Increment("me", 2 to 3)), id = 1))
+
+    CrdtCount.Data()
+    receiverCaptor.lastValue(
+      CrdtCount.Data(
+        mutableMapOf("me" to 1, "them" to 1),
+        VersionMap("me" to 1, "them" to 1)
+      ),
+      1
+    )
+    receiverCaptor.lastValue(
+      CrdtCount.Data(
+        mutableMapOf("me" to 1, "them" to 2),
+        VersionMap("me" to 1, "them" to 2)
+      ),
+      2
+    )
+
+    activeStore.idle()
+
+    assertThat(activeStore.localData).isEqualTo(driverModelCaptor.lastValue)
+  }
+
+  private suspend fun setupMocks(): Pair<Driver<CrdtCount.Data>, DriverProvider> {
```

Comment:
Any reason not to make this an `@Before` method?

Reply:
Unfortunately, in order to mock the `send` and other `suspend fun`s, they need to be created within a suspend function (or in a coroutine), and I thought that might be a bit sketchy within `@Begin`